alacritty_terminal/src/term/mod.rs

```diff
@@ -819,8 +1072,17 @@ impl<T> Term<T> {
         cursor_cell.flags = flags;
     }
 
-    pub fn colors(&self) -> &Colors {
-        &self.colors
+    #[inline]
+    fn damage_cursor(&mut self) {
```

Comment:
> Moving last_cursor around gets complicated when you start moving it back and forth. I think I'd rather do simpler, but in a way that you can't really fail with collecting damage.

Why do we need to move this around? Would it not be possible to reuse the `last_cursor` on `self.damage` as-is?

Would it not be sufficient to always damage the delta to the last cursor position when checking it? Something like this:

```
let last_cursor = mem::replace(&mut self.damage.last_cursor, self.grid.cursor.point);
self.damage.damage_point(last_cursor);
```

Since we always damage the current cursor state at least, this should damage the cursor at every stage of the process, right? Or are there scenarios where the cursor is moved, but damage is skipped? If so that would surely create issues wrt `input`.

Reply:
This won't work. Consider the following. You can't use `last_cursor` since it's not accurate, because it doesn't account for changes from `fn input`. Consider the following:


```rust
// You have no damage at all.
term.goto(1,1); // last_cursor is on 0,0, so damaging it and setting to 1,1.
term.input('a');
term.input('b');
term.input('c');
term.goto(2,0); // Damaging 1,1 and setting to 2,0.
// We failed to damaged line properly...
```

And such case will be in other functions where you change the line. So if we want to use last_cursor position we will have to damage it twice here as well, but It just proves that there's no need to carry last_cursor position around, since doing it simpler is better.

In function where cursor can't move to other lines I'm damaging line, not cursor.